test_py2.py

```diff
@@ -0,0 +1,102 @@
+import pytest
+import sys
+
+import semver
+
+
+PY2 = sys.version_info[0] == 2
+PY3 = sys.version_info[0] == 3
+
+
+def ensure_binary(s, encoding="utf-8", errors="strict"):
+    """Coerce **s** to six.binary_type.
+
+    For Python 2:
+      - `unicode` -> encoded to `str`
+      - `str` -> `str`
+
+    For Python 3:
+      - `str` -> encoded to `bytes`
+      - `bytes` -> `bytes`
+    """
+    if isinstance(s, semver.text_type):
+        return s.encode(encoding, errors)
+    elif isinstance(s, semver.binary_type):
+        return s
+    else:
+        raise TypeError("not expecting type '%s'" % type(s))
+
+
+def test_should_work_with_string_and_unicode():
+    result = semver.compare(semver.u("1.1.0"), semver.b("1.2.2"))
+    assert result == -1
+    result = semver.compare(semver.b("1.1.0"), semver.u("1.2.2"))
+    assert result == -1
+
+
+class TestEnsure:
+    # From six project
+    # grinning face emoji
+    UNICODE_EMOJI = semver.u("\U0001F600")
+    BINARY_EMOJI = b"\xf0\x9f\x98\x80"
```

Comment:
I was thinking that you wouldn't be able to do this in a single file because Python 2.x would not recognize the `b"..."` syntax. But, looking at the docs again (since I'm really not that expert in Python), I see that Py2 doesn't consider `b"..."` to be an error; it just ignores the `b` and treats it as a regular string. So I guess this will work... but I do find it somewhat confusing at first glance, so you might want to add a comment to clarify that this code is being interpreted very differently in Py2 vs. Py3.

Reply:
I also find it mildly confusing that this file is called `test_py2.py` even though it gets run in both Python 2 and Python 3. I understand that the point is to verify that it behaves consistently between the two with equivalent inputs, I'm just not sure that that would be clear to someone who was looking at this code without having seen our previous conversation. But then again I'm a little sleepy now and so I find lots of things mildly confusing.

Reply:
But I really should emphasize that I'm not super adept at Python. You may want to run this PR by someone else who's had to deal with a lot of Py2/Py3 migration issues because I'm not sure I would catch any real problems.

Reply:
> recognize the `b"..."` syntax. [...] So I guess this will work... but I do find it somewhat confusing at first glance, so you might want to add a comment to clarify that this code is being interpreted very differently in Py2 vs. Py3.

Yes, I've got it from the six package which deals with Py2 vs. Py3 compatibility a lot. :wink: 


> I also find it mildly confusing that this file is called `test_py2.py`

Yes, I will renamed it to `test_typeerror-274.py`. It's probably not ideal either, but it gives a reference to this issue.